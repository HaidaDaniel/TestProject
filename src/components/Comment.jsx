/** @format */
import PropTypes from 'prop-types'

import { RatingStars } from './index'

import {
  StyledCard,
  StyledCardBody,
  StyledCardHeader,
  StyledCardText,
  StyledAuthorText,
} from '../styled/CommentStyles'

const Comment = ({ author, text, rating }) => (
  <StyledCard>
    <StyledCardHeader>
      <RatingStars rating={rating} isInput={false} />
    </StyledCardHeader>
    <StyledCardBody>
      <StyledCardText>{text}</StyledCardText>
      <StyledAuthorText>{author}</StyledAuthorText>
    </StyledCardBody>
  </StyledCard>
)

export default Comment

Comment.propTypes = {
  author: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
}
