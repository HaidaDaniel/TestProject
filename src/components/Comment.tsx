/** @format */
import { FC } from 'react'

import { RatingStars } from './index'

import {
  StyledCard,
  StyledCardBody,
  StyledCardHeader,
  StyledCardText,
  StyledAuthorText,
} from '../styled/CommentStyles'
import { IComment } from './interfaces/IProductData'

const Comment: FC<IComment> = ({ author, text, rating, commentId }) => (
  <StyledCard>
    <StyledCardHeader>
      <RatingStars rating={rating} isInput={false} />
    </StyledCardHeader>
    <StyledCardBody>
      <StyledCardText>{text}</StyledCardText>
      <StyledAuthorText>
        author:{author} <br /> review id:{commentId}
      </StyledAuthorText>
    </StyledCardBody>
  </StyledCard>
)

export default Comment
