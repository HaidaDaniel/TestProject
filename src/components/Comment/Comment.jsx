/** @format */
import { Card, Row, Col } from 'react-bootstrap'
import RatingStars from '../RatingStars/RatingStars'
import './index.css'

function Comment({ author, text, rating }) {
  return (
    <Card style={{ width: '100%' }}>
      <Card.Body>
        <Card.Header className='d-flex'>
          <RatingStars rating={rating} />
        </Card.Header>
        <Card.Subtitle className='mb-2 text-muted'></Card.Subtitle>Name:{' '}
        {author}
        <Card.Text className='d-flex '>{text}</Card.Text>
      </Card.Body>
    </Card>
  )
}

export default Comment
