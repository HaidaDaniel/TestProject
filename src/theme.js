const theme = {
    colors: {
        primary: '#0d6efd',
        secondary: '#6c757d',
        success: '#28a745',
        error: '#dc3545',
        golden: '#f5a623'
    },
    fonts: {
        body: 'Arial, sans-serif',
        heading: 'Helvetica, sans-serif',
    },

};

export default theme;