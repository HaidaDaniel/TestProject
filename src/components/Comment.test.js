
import React from 'react'
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';

import theme from '../theme';
import Comment from './Comment';

describe('Comment Component', () => {
    it('renders comment details correctly', () => {
        const comment = {
            author: 'John',
            text: 'Test comment',
            rating: 5,

        };

        const { getByText } = render(
            <ThemeProvider theme={theme}>
                <Comment
                    author={comment.author}
                    text={comment.text}
                    rating={comment.rating}

                /></ThemeProvider>
        );


        expect(screen.getByText((content, element) => {

            return content.includes('author:') && content.includes('John');
        })).toBeInTheDocument();
        expect(screen.getByText((content, element) => {

            return content.includes('review id');
        })).toBeInTheDocument();
        expect(screen.getByText(comment.text)).toBeInTheDocument();

        expect(document.body).toMatchSnapshot();

    });
});  