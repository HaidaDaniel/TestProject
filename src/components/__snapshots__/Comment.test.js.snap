// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component renders comment details correctly 1`] = `
<body>
  <div>
    <div
      class="sc-gXCJSa iWMtLk card"
    >
      <div
        class="sc-ciJnBw dvZznk card-header"
      >
        <div
          class="sc-ezGUZh bxdEeg"
        >
          <div
            class="sc-hhWzdI RnQUK"
          >
            <span
              class="sc-bYUneI sc-kMbQoj eLouB doquPp"
            >
              ★
            </span>
          </div>
          <div
            class="sc-hhWzdI RnQUK"
          >
            <span
              class="sc-bYUneI sc-kMbQoj eLouB doquPp"
            >
              ★
            </span>
          </div>
          <div
            class="sc-hhWzdI RnQUK"
          >
            <span
              class="sc-bYUneI sc-kMbQoj eLouB doquPp"
            >
              ★
            </span>
          </div>
          <div
            class="sc-hhWzdI RnQUK"
          >
            <span
              class="sc-bYUneI sc-kMbQoj eLouB doquPp"
            >
              ★
            </span>
          </div>
          <div
            class="sc-hhWzdI RnQUK"
          >
            <span
              class="sc-bYUneI sc-kMbQoj eLouB doquPp"
            >
              ★
            </span>
          </div>
          5
           
          /5
        </div>
      </div>
      <div
        class="sc-cCIIiH dCQhKt card-body"
      >
        <p
          class="sc-jdkVqZ kebNMo card-text"
        >
          Test comment
        </p>
        <p
          class="sc-bZCZoC yMTIK card-text"
        >
          author:
          John
           
          <br />
           review id:
        </p>
      </div>
    </div>
  </div>
</body>
`;
